### PR TITLE
week-api add research afternoon to api week

### PR DESCRIPTION
**Pull request by lucyrose93**
*NB. Even though this topic is still up for discussion (see #450), I have created a PR because this issue relates to the schedule for tomorrow, so time is precious.*

We are proposing to add a research afternoon to API week, based on Elias' workshop, for reasons detailed in issue #450. 

The research topics for this research afternoon will cover all of the topics in the [original workshop](https://github.com/foundersandcoders/ws-software-design-js) we were intending to deliver. 

@jsms90 @bradreeder @eliascodes What do you think about adding this research afternoon?

Relates #450 & relates #300

### CODE REVIEW

File: coursebook/week-3/research-afternoon.md [hunk 1]
```diff
@@ -1,3 +1,44 @@
...
+A [resource](https://github.com/foundersandcoders/ws-software-design-js/tree/master/stage-1/closures-and-scope) if you get stuck!
+
+## Abstraction with Functions
+- What is abstraction (in relation to code)?
+- How can abstraction be useful?
+- Attempt this [exercise](https://github.com/foundersandcoders/ws-software-design-js/blob/master/stage-1/abstraction-with-functions/exercise.js) and let us know how you get on!
+
+A [resource](https://github.com/foundersandcoders/ws-software-design-js/tree/master/stage-1/abstraction-with-functions) if you get stuck.
+
+##The Module pattern
```
Comment (m4v15) [inline coursebook/week-3/research-afternoon.md]: Hey sorry, was just looking at this for next week in Naz and noticed you need a space here for your mark down to work.

**Comment by eliascodes** [comment]
So I've left a comment in #450 with my initial thoughts. You've set aside 1.5 hours for the research period; given that we link to resources that basically give them the answers, that's probably enough, but you may want to stay flexible on it.

I have no blocking comments, so once @m4v15 comment is addressed, I can approve.

I would like some feedback on how this way of delivering the material went, though.

**Comment by jsms90** [comment]
I think this was a mistake? @lucyrose93 @smarthutza the flexbox workshop only takes 45 mins - 1 hour, right?

So it would be
10-11am flexbox
11am-1pm software design workshop
-- lunch --
research afternoons

File: coursebook/week-3/research-afternoon.md (hunk 1 above)
Comment (lucyrose93) [inline coursebook/week-3/research-afternoon.md]: Nice spot @m4v15
Have added the space.

**Comment by lucyrose93** [comment]
**Thanks @jsms90**. I believe @smarthutza actually wanted to extend it to 1hr45 to allow for more time to be spent on CSS Tricks and Flexbox Froggy. These resources are already in the workshop repo, but @smarthutza you may need to update your readme for future cohorts to explicitly state how much time is spent on each activity. 

**@eliascodes** I will let you know how it goes tomorrow. I'm hoping that they can take the material and present it in their own way. Thanks for the review.

**Comment by lucyrose93** [comment]
@eliascodes @jsms90 
So @JWLD has just let me know that he would be happy to deliver Elias' workshop as originally intended. It's a bit of a pivot but Jack volunteered to help on Sunday and has just gone through the workshop. He is a lot more confident than I am in explaining how to implement the concepts, so I think we can make this work after all. 
He says 2.5 hours is a good amount of time, so that's perfect for our schedule.

**Comment by lucyrose93** [comment]
am closing this as we're sticking to the old plan -- #450

**Pull request closed without merging**